secrets/create-secrets.sh

```diff
@@ -33,14 +33,23 @@ then
   kubectl -n "$NAMESPACE" delete secret spin-secrets
 fi
 
+quoteRequiredRegex="[[:space:]]+"
 SEC_FILE=$CWD/secrets.env
 [[ ! -f $SEC_FILE  ]] && echo "WARNING: Missing file $SEC_FILE, loading sample secrets from $CWD/secrets-example.env instead" && SEC_FILE=$CWD/secrets-example.env
-while IFS= read -r l
+while IFS='=' read -r literalsLine; 
 do
-  echo "Including secret literal \"$(echo $l | sed 's|=.*||')\""
-  LITERALS+="--from-literal=$l "
-done < <(grep -v -e '^#' -e '^$' < "$SEC_FILE")
-LITERALS=${LITERALS::${#LITERALS}-1}
+    echo "Including secret literal \"$(echo $literalsLine)\""
+    literalsKey="${literalsLine%%=*}"
+    echo "key is ${literalsKey}"
+    literalsValue="${literalsLine#*=}"
+    echo "val is ${literalsValue}"
```

Comment:
Do you think it's necessary to print the secret value or this is for debugging the script?

Reply:
Correct, I figured since the script kind of did some things for you (like putting double quotes around spaces) it could be useful for it to tell you what it is trying to put into the secret for debugging purposes because in-reality the script will probably be run quite rarely so it's not like the debugging information will be seen all day every day.

Reply:
I can just comment them out so that it's easy for someone to uncomment if the need arises